assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/llava/utils.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Mlflow LLaVA wrapper helper functions."""
+
+import base64
+import re
+import requests
+
+from typing import Union
+
+
+def process_image(image: Union[str, bytes]) -> bytes:
+    """Process input image.
+
+    If input image is in bytes format, return it as it is.
+    If input image is in base64 string format, decode it to bytes.
+    If input image is in url format, download it and return bytes.
+    https://github.com/mlflow/mlflow/blob/master/examples/flower_classifier/image_pyfunc.py
+
+    :param image: image in base64 string format or url or bytes.
+    :type image: string or bytes
+    :return: decoded image.
+    :rtype: bytes
+    """
+    if isinstance(image, bytes):
+        return image
+    elif isinstance(image, str):
+        if _is_valid_url(image):
+            return requests.get(image).content
+        else:
+            try:
+                return base64.b64decode(image)
+            except ValueError:
+                raise ValueError(
+                    "The provided image string cannot be decoded. Expected format is Base64 or URL string."
+                )
+    else:
+        raise ValueError(
+            f"Image received in {type(image)} format which is not supported."
+            "Expected format is bytes, base64 string or url string."
+        )
+
+
+def _is_valid_url(text: str) -> bool:
```

Comment:
This looks very similar to the function from common/vision_utils.py. Can we update that function with the additional changes and use that directly here.

Reply:
Done. As we discussed, also added unit tests but checking things E2E will be up to model owners.